File: src/main/java/com/example/junit/test/controller/MainController.java
package com.example.junit.test.controller;

import com.example.junit.test.service.MainService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("")
public class MainController {

    @Autowired
    private MainService mainService;

    @GetMapping("/test")
    public boolean check(@RequestParam(value = "str") String str){
        return mainService.checkString(str);
    }

}

File: kafkaclient/src/main/kotlin/com/example/kafkaclient/MainController.kt
package com.example.kafkaclient

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("")
class MainController {

    @Autowired
    private val kafkaTemplate: KafkaTemplate<String, String>? = null

    @GetMapping("/order")
    fun order(@RequestParam("apple") apple: Int,
              @RequestParam("orange") orange: Int) {
        var appleNum = apple
        var orangeNum = orange

        val msg = StringBuilder()
        while (appleNum > 0) {
            msg.append("apple ")
            appleNum--
        }
        while (orangeNum > 0) {
            msg.append("orange ")
            orangeNum--
        }
        kafkaTemplate?.send("topic", msg.toString())
    }

}
